osu.Game/Screens/Play/HUD/HitErrorMeters/ColourHitErrorMeter.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Scoring;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.Play.HUD.HitErrorMeters
+{
+    public class ColourHitErrorMeter : HitErrorMeter
+    {
+        private const int max_available_judgements = 20;
+
+        private readonly JudgementFlow judgementsFlow;
+        private readonly BindableList<(Color4 colour, JudgementResult result)> judgements = new BindableList<(Color4 colour, JudgementResult result)>();
+
+        public ColourHitErrorMeter(HitWindows hitWindows)
+            : base(hitWindows)
+        {
+            AutoSizeAxes = Axes.Both;
+            InternalChild = judgementsFlow = new JudgementFlow();
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            judgementsFlow.Judgements.BindTo(judgements);
+        }
+
+        public override void OnNewJudgement(JudgementResult judgement)
+        {
+            judgements.Add((GetColourForHitResult(HitWindows.ResultFor(judgement.TimeOffset)), judgement));
+
+            if (judgements.Count > max_available_judgements)
+                judgements.RemoveAt(0);
+        }
+
+        private class JudgementFlow : Container<DrawableResult>
```

Comment:
is there a reason you didn't use `FillFlow` for this?

Reply:
Just the first solution I came up with

Reply:
you are reimplementing fillflow logic. try using a fillflow instead.

Reply:
I guess, I should look into NotificationsOverlay for similar behavior